src/parser/ParsedQuery.h

```diff
@@ -409,9 +409,18 @@ class ParsedQuery {
     return std::get<ConstructClause>(_clause);
   }
 
+  // Add a variable, that was found in the SubQuery body, when query has a
+  // Select Clause
+  [[nodiscard]] bool addVariableFromSubQueryBody(ParsedQuery* query,
+                                                 const string& variable) const {
+    if (!query->hasSelectClause()) return false;
+    if (query->selectClause()._varsOrAsterisk.isAsterisk()) {
+      query->selectClause()._varsOrAsterisk.addVariableFromQueryBody(variable);
+    }
+    return true;
+  }
+
```

Comment:
Why does this function take the ParsedQuery as a pointer? Just make it a normal member function with the `variable` as a single argument. The it is not const, and not `[[nodiscard]]`.

Also it does not only add variables from a subquery body , but mostly from the query body.
I would suggest the name `registerVariableVisibleInQueryBody` and also rename the member function of `VarsOrAsterisk` like this.